tsl/src/continuous_aggs/create.c

```diff
@@ -1779,36 +1798,32 @@ cagg_update_view_definition(ContinuousAgg *agg, Hypertable *mat_ht,
 	Query *user_query = get_view_query(user_view_rel);
 	relation_close(user_view_rel, AccessShareLock);
 
-	Oid direct_view_oid = relation_oid(agg->data.direct_view_schema, agg->data.direct_view_name);
-
-	Relation direct_view_rel = relation_open(direct_view_oid, AccessShareLock);
-	Query *direct_query = copyObject(get_view_query(direct_view_rel));
-
-	CAggTimebucketInfo timebucket_exprinfo = cagg_validate_query(direct_query);
-
-	relation_close(direct_view_rel, AccessShareLock);
 	FinalizeQueryInfo fqi;
 	MatTableColumnInfo mattblinfo;
 	ObjectAddress mataddress = {
 		.classId = RelationRelationId,
 		.objectId = mat_ht->main_table_relid,
 	};
 
+	Oid direct_view_oid = relation_oid(agg->data.direct_view_schema, agg->data.direct_view_name);
+	Relation direct_view_rel = relation_open(direct_view_oid, AccessShareLock);
+	Query *direct_query = copyObject(get_view_query(direct_view_rel));
+	List *rtable = direct_query->rtable;
+	/* When a view is created (StoreViewQuery), 2 dummy rtable entries corresponding to "old" and
+	 * "new" are prepended to the rtable list. We remove these and adjust the varnos to recreate
+	 * the original user supplied direct query
+	 */
+	rtable = list_delete_first(rtable);
```

Comment:
can we have some asserts here that rtable is actually in the shape we assume

Reply:
added.